CI fails because of a DeprecationWarning from `pandas` regarding `pyarrow`

CI fails because of a DeprecationWarning from `pandas`

```python
E   DeprecationWarning: 
E   Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
E   (to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
E   but was not found to be installed on your system.
E   If this would cause problems for you,
E   please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
```

see https://github.com/pandas-dev/pandas/issues/54466

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -72,7 +72,7 @@ minversion = "6.0"
 addopts = ["-ra", "--showlocals", "--strict-markers", "--strict-config"]
 xfail_strict = true
 filterwarnings = [
-  "error",
+  "error", "ignore::DeprecationWarning",
 ]
 log_cli_level = "INFO"
 testpaths = [

